peekingduck/pipeline/nodes/draw/utils/legend.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2021 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+functions for drawing legend related UI components
+"""
+
+from typing import Dict, List, Any
+import numpy as np
+import cv2
+from cv2 import FONT_HERSHEY_SIMPLEX, LINE_AA
+from peekingduck.pipeline.nodes.draw.utils.constants import \
+    THICK, WHITE, SMALL_FONTSCALE, BLACK, FILLED
+from peekingduck.pipeline.nodes.draw.utils.general import get_image_size
+
```

Comment:
Just a thought - as we add more nodes in the future, others are going to add code to this file. Are there any nuances you've encountered here that should be included here as comments?

Reply:
Hmmm... I will put steps here as a guide to what to do:
If a new draw node that draws a new line onto the legends box, individuals should:
1) use the name of the key for the input required and add it to all_legend_items
2) Add function into script, and add it into the get_legend_registry function, using the same key name
3) Manipulate the _get_legend_height function if there are unique rules to the increase in legend box height

In the long run, if this also becomes a huge chunk, we might wanna relook at the restructuring and introduction of a HUD design for all the different infos, as we discussed.

Reply:
Okay sounds good!